sklearn/decomposition/fastica_.py

```diff
@@ -528,22 +531,29 @@ def fit(self, X, y=None):
         self._fit(X, compute_sources=False)
         return self
 
-    def transform(self, X, y=None, copy=True):
+    def transform(self, X, y='deprecated', copy=True):
```

Comment:
because here there is copy param so we cannot just remove y from the signature like above.